draft-ietf-jsonpath-base.md

```diff
@@ -1108,17 +1108,23 @@ Since bracket notation is more general than dot notation, it is used to construc
 the syntax of Normalized Paths is restricted so that there is one and only one way of representing any
 given Normalized Path.
 
+Single quotes are used to delimit string member names. This makes it possible to use Normalized Paths
+as strings in JSON documents (delimited with double quotes) without needing to escape the contents of
```

Comment:
That is not true.  You still need to escape some of the contents of the strings when putting them into JSON; you just get `'` for free (and stay reasonably readable that way).

Reply:
`"`, I think.

Reply:
Yes of course that must be done manually ... but it is very unlikely to meet member names in any JSON document with embedded `"` (bad behavior).

Reply:
what about pragmatically reusing the same ABNF, that rfc8259 uses for member names ... adding unescaped `'` of course ?

Reply:
I think we already do...

```
      member = string name-separator value
```

where "string" is called "string-literal".
RFC 8259 hasn't done the work to avoid unpaired surrogates, so going back to their definition of "string" would be a regression.

But maybe I don't understand what you are trying to do.

Reply:
oh ... I would like to mention the fact somewhere, that normalized pathes should be naturally the most performant queries in any implementation, compared with other - more complex - JSONPath queries leading to the same result node.

Reply:
> But maybe I don't understand what you are trying to do.

you did ... thanks for the clarification.

Reply:
You're correct that 8259 doesn't rule out naked surrogates and suchlike because the consensus was we couldn't retroactively ban things that were previously JSON per previous specifications. That's what I-JSON is for.  So the question is, should jsonpath member names conform to JSON or I-JSON?

Reply:
I-JSON.

Reply:
8259 is schizophrenic: It requires UTF-8 for unescaped, but doesn't exclude unpaired surrogates for escaped.  That clearly would have been fixed if we had been a bit more audacious at the time.

Reply:
OK, whether jsonpath should be more restrictive than JSON in what names it allows is an issue we're going to have to take up on its own.

Reply:
As a WG chair, you may have to say something like this.

From a technical point of view, I find the suggestion that we could do anything that could be understood to lend additional credibility to the Zombie that Section 8.2 of RFC 8259 is, simply untenable. That isn't JSON, it is crimes of the past that wander the unlit cemeteries waiting to finally be buried.